src/mainboard/protectli/vault_ehl/devicetree.cb

```diff
@@ -24,86 +22,175 @@ chip soc/intel/elkhartlake
 
 	register "tcc_offset" = "5" # TCC of 95C
 
-
-	register "usb2_ports[0]" = "USB2_PORT_SHORT(OC_SKIP)"	# USB3.0 Type A
-	register "usb2_ports[1]" = "USB2_PORT_SHORT(OC_SKIP)"	# USB3.0 Type A
-	register "usb2_ports[2]" = "USB2_PORT_SHORT(OC_SKIP)"	# USB3.0 Type C
-	register "usb2_ports[3]" = "USB2_PORT_SHORT(OC_SKIP)"	# M.2 WiFi
-	register "usb2_ports[4]" = "USB2_PORT_SHORT(OC_SKIP)"	# mPCIe LTE
+	# USB 2.0 ports
+	register "usb2_ports[0]" = "USB2_PORT_SHORT(OC_SKIP)"	# Header FUSB1
+	register "usb2_ports[1]" = "USB2_PORT_EMPTY"
+	register "usb2_ports[2]" = "USB2_PORT_SHORT(OC_SKIP)"	# Header FUSB1
+	register "usb2_ports[3]" = "USB2_PORT_SHORT(OC_SKIP)"	# M.2 WLAN
+	register "usb2_ports[4]" = "USB2_PORT_SHORT(OC_SKIP)"	# M.2 WWAN
 	register "usb2_ports[5]" = "USB2_PORT_EMPTY"
-	register "usb2_ports[6]" = "USB2_PORT_EMPTY"
-	register "usb2_ports[7]" = "USB2_PORT_EMPTY"
-	register "usb2_ports[8]" = "USB2_PORT_EMPTY"
+	register "usb2_ports[6]" = "USB2_PORT_SHORT(OC_SKIP)"	# USB Type-A Upper
+	register "usb2_ports[7]" = "USB2_PORT_SHORT(OC_SKIP)"	# USB Type-C
+	register "usb2_ports[8]" = "USB2_PORT_SHORT(OC_SKIP)"	# USB Type-A Lower
 	register "usb2_ports[9]" = "USB2_PORT_EMPTY"
 
-	register "usb3_ports[0]" = "USB3_PORT_DEFAULT(OC_SKIP)"	# USB3.0 Type A
-	register "usb3_ports[1]" = "USB3_PORT_DEFAULT(OC_SKIP)"	# USB3.0 Type A
-	register "usb3_ports[2]" = "USB3_PORT_DEFAULT(OC_SKIP)"	# USB3.0 Type C
+	# USB 3.x ports
+	register "usb3_ports[0]" = "USB3_PORT_DEFAULT(OC_SKIP)"	# USB Type-A Upper
+	register "usb3_ports[1]" = "USB3_PORT_DEFAULT(OC_SKIP)"	# USB Type-A Lower
+	register "usb3_ports[2]" = "USB3_PORT_DEFAULT(OC_SKIP)"	# USB Type-C Muxed
 	register "usb3_ports[3]" = "USB3_PORT_EMPTY"
 
+	# PCIe root ports related UPDs
+	register "PcieRpEnable[1]" = "1"
+	register "PcieRpEnable[2]" = "1"
+	register "PcieRpEnable[3]" = "1"
+	register "PcieRpEnable[4]" = "1"
+	register "PcieRpEnable[6]" = "1"
+
+	register "PcieClkSrcUsage[0]" = "PCIE_CLK_NOTUSED"
+	register "PcieClkSrcUsage[1]" = "PCIE_CLK_FREE"
+	register "PcieClkSrcUsage[2]" = "PCIE_CLK_FREE"
+	register "PcieClkSrcUsage[3]" = "PCIE_CLK_FREE"
+	register "PcieClkSrcUsage[4]" = "PCIE_CLK_FREE"
+	register "PcieClkSrcUsage[5]" = "PCIE_CLK_FREE"
+
+	register "PcieClkSrcClkReq[0]" = "PCIE_CLK_NOTUSED"
+	register "PcieClkSrcClkReq[1]" = "PCIE_CLK_NOTUSED"
+	register "PcieClkSrcClkReq[2]" = "PCIE_CLK_NOTUSED"
+	register "PcieClkSrcClkReq[3]" = "PCIE_CLK_NOTUSED"
+	register "PcieClkSrcClkReq[4]" = "PCIE_CLK_NOTUSED"
+	register "PcieClkSrcClkReq[5]" = "PCIE_CLK_NOTUSED"
+
+	# Storage (SATA/SDCARD/EMMC) related UPDs
+	register "SataSalpSupport" = "1"
+	register "SataPortsEnable[0]" = "1" # Header
+	register "SataPortsEnable[1]" = "1" # M.2 2280
+	register "SataPortsDevSlp[0]" = "0"
+	register "SataPortsDevSlp[1]" = "1"
+
```

Comment:
PchHdaAudioLinkHdaEnable = 1 ?
PchHdaSdiEnable[0] = 1 ?
ScsEmmcHs400Enabled = 1 ?
SkipCpuReplacementCheck = 1 ?

DdiPortAConfig ? 
DdiPortBConfig ?
DdiPort*Hpd ?
DdiPort*Ddc ?

Reply:
Added DDC / HPD params for DdiPortA and DdiPortC, the rest aren't wired up in soc code

Reply:
Then we have to wire it up.

Reply:
done

Reply:
Good. Except that DDIPortC has no HDP and DDC connected on the schematics (this is the USB-C DP)

Reply:
Removed HPD/DDC config for Port C.